src/sql/parts/dashboard/tabs/dashboardInnerTab.contribution.ts

```diff
@@ -0,0 +1,111 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import { IExtensionPointUser, ExtensionsRegistry } from 'vs/platform/extensions/common/extensionsRegistry';
+import { IJSONSchema } from 'vs/base/common/jsonSchema';
+import { localize } from 'vs/nls';
+import { join } from 'path';
+import { createCSSRule } from 'vs/base/browser/dom';
+import URI from 'vs/base/common/uri';
+
+import { registerInnerTab, generateInnerTabContentSchemaProperties } from 'sql/platform/dashboard/common/innerTabRegistry';
+
+export type IUserFriendlyIcon = string | { light: string; dark: string; };
+
+export interface IDashboardInnerTabContrib {
+	id: string;
+	title: string;
+	icon?: IUserFriendlyIcon;
+	content: object;
+}
+
+const innerTabSchema: IJSONSchema = {
+	type: 'object',
+	properties: {
+		id: {
+			type: 'string',
+			description: localize('sqlops.extension.contributes.dashboard.innertab.id', "Unique identifier for this inner tab. Will be passed to the extension for any requests.")
+		},
+		icon: {
```

Comment:
Excellent - so we can provide an icon? If so how does the text vs icon handling work? It seems like we should consider enforcing a standard icon size so things look "right" just as we have for the viewlet icons.

Essentially I'm wondering if we should enforce an icon and have title appear on hover instead.

Reply:
Panel hasn't support icon yet. This is the same icon schema that VSCode use for command icon. 

---
In reply to: [169769616](https://github.com/Microsoft/sqlopsstudio/pull/751#discussion_r169769616) [](ancestors = 169769616)